src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -10,5 +10,13 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
   if (typesUsedInValue.size > 1) {
     throw new UnsupportedOperationException(s"Cannot construct array with a mixed types: $value")
   }
-  override def toWdlString(): String = s"[${value.map{x=>x.toWdlString}.mkString(", ")}]"
+
+  override def toWdlString(): String = s"[${value.map { x => x.toWdlString }.mkString(", ")}]"
+
+  def map[R <: WdlValue](f: WdlValue => R): WdlArray = {
+    value.map{f} match {
+      case s:Seq[R] if s.nonEmpty => WdlArray(WdlArrayType(s.head.wdlType), s)
+      case _ => this
```

Comment:
Wait, be careful with that. I think it "works" with Seqs as in if it's actually a List it'll work but if you pass in something else it won't.